pandas/tests/copy_view/test_setitem.py

```diff
@@ -79,13 +79,54 @@ def test_set_columns_with_dataframe(using_copy_on_write):
     df[["c", "d"]] = df2
 
     if using_copy_on_write:
-        # TODO(CoW) with CoW we can delay the copy
-        # assert np.shares_memory(df["c"].values, df2["c"].values)
-        assert not np.shares_memory(df["c"].values, df2["c"].values)
+        assert np.shares_memory(df["c"].values, df2["c"].values)
     else:
         # the data is copied
         assert not np.shares_memory(df["c"].values, df2["c"].values)
 
     # and modifying the set DataFrame does not modify the original DataFrame
     df2.iloc[0, 0] = 0
     tm.assert_series_equal(df["c"], Series([7, 8, 9], name="c"))
+
+
+def test_setitem_series_no_copy(using_copy_on_write):
```

Comment:
How is this test different from the other ones above that you edited?

Reply:
Missed that, but there is actually a difference, the existing test is a bit buggy, I will copy this one over and modify the other one as appropriate

Reply:
Ah no I keep it here and modify the other test.

Reply:
Regarding the other test: As soon as you update ``s``, a copy is triggered and updating the df afterwards never has an effect if the initial copy worked because they don't share data anymore. The df-update is now handled by the new test. 

Regarding the different cases: overwriting an existing column takes a different code path than adding a new one, so we have to test both